packages/gatsby/src/utils/webpack.config.js

```diff
@@ -634,12 +635,21 @@ module.exports = async (
         framework: {
           chunks: `all`,
           name: `framework`,
-          // This regex ignores nested copies of framework libraries so they're bundled with their issuer.
-          test: new RegExp(
-            `(?<!node_modules.*)[\\\\/]node_modules[\\\\/](${FRAMEWORK_BUNDLES.join(
-              `|`
-            )})[\\\\/]`
-          ),
+          test: module => {
+            // Packages like gatsby-plugin-image might import from "react-dom/server". We don't want to include react-dom-server in the framework bundle.
+            // A rawRequest might look like these:
+            // - "react-dom/server"
+            // - "node_modules/react-dom/cjs/react-dom-server-legacy.browser.production.min.js"
+            // Use a "/" before "react-dom-server" so that we don't match packages that contain "react-dom-server" in their name
+            if (
+              module.rawRequest === `react-dom/server` ||
+              module.rawRequest.includes(`/react-dom-server`)
+            ) {
+              return false
+            }
+
+            return FRAMEWORK_BUNDLES_REGEX.test(module.nameForCondition())
```

Comment:
We seem to use `module.identifier()` in other places but most often on the internet I've seen people use `module.nameForCondition()`

Reply:
it doesn't really matter for this use case. nameForCondition returns the resource without any query parts.
https://github.com/webpack/webpack/blob/293e677b355da0d5bc4ddfc97d2afec114912711/lib/NormalModule.js#L375-L380

Reply:
this change brakes _gatsby-plugin-preact_ which [assumes that `test` function is a regex](https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-plugin-preact/src/gatsby-node.js#L56)